aospy/model.py

```diff
@@ -16,16 +15,181 @@
+    # Clean up: correct names and dimension order.
+    sfc_area = sfc_area.rename(internal_names.SFC_AREA_STR)
+    sfc_area[internal_names.LAT_STR] = lat
+    sfc_area[internal_names.LON_STR] = lon
+    return sfc_area.transpose()
+
+
 class Model(object):
-    """Parameters of local data associated with a single climate model."""
-    def __init__(self, name='', description='', proj=None,
+    """An object that describes a single climate or weather model.
+
+    Each `Model` object is associated with a parent `Proj` object and also with
+    one or more child `Run` objects.
+
+    If aospy is being used to work with non climate- or weather-model data, the
+    `Model` object can be used e.g. to represent a gridded observational
+    product, with its child `Run` objects representing different released
+    versions of that dataset.
+
+    Attributes
+    ----------
+    name : str
+        The model's name
+    description : str
+        A description of the model
+    proj : {None, aospy.Proj}
+        The model's parent aospy.Proj object
+    runs : dict
+        A dictionary with entries of the form ``{run_obj.name: run_obj}``,
+        for each of this model's child Run objects
+    default_runs : dict
+        The default subset of chidl run objects on which to perform
+        calculations via `aospy.Calc` with this model if not otherwise
+        specified
+    grid_file_paths : list
+        The paths to netCDF files stored on disk from which the model's
+        coordinate data can be taken.
+    default_start_date, default_end_date : datetime.datetime
+        The default start and end dates of any calculations using this Model
+    data_loader : aospy.DataLoader
+        The aospy.DataLoader object used to find data on disk corresponding
+        to this Model object
+
+    """
+
+    def __init__(self, name=None, description=None, proj=None,
                  grid_file_paths=None, default_start_date=None,
-                 default_end_date=None,
-                 runs={}, default_runs={},
-                 load_grid_data=False,
-                 data_loader=None):
-        self.name = name
-        self.description = description
+                 default_end_date=None, runs=None, default_runs=None,
+                 load_grid_data=False, data_loader=None):
+        """
+        Parameters
+        ----------
+        name : str
+            The model's name.  This must be unique from that of any other
+            `Model` objects being used by the parent `Proj`.
+        description : str, optional
+            A description of the model.  This is not used internally by
+            aospy; it is solely for the user's information.
+        proj : {None, aospy.Proj}, optional
+            The parent Proj object.  When the parent `Proj` object is
+            instantiated with this Model included in its `models` attribute,
+            this will be over-written with that `Proj` object.
+        data_loader : aospy.DataLoader
```

Comment:
I *think* the way things are set up now prevents this ([see this comment in #97](https://github.com/spencerahill/aospy/pull/97#issuecomment-263038018)):

The issue currently has to do with a circular dependency (the constructor of the model object depends on Run objects already being instantiated), so one cannot reference that template (defined in the Model), when constructing the Run.  In order to leverage a "template" DataLoader right now, one would have to first create the template DataLoader where the Runs are created, and then use the alternative constructor for a GFDLDataLoader (the only one that supports this currently) that takes a template DataLoader as its first argument in each of the Runs that "inherit" it.

> So what's the point of having a model-level (or proj-level) DataLoader at all?

In hindsight, yes, it doesn't seem to make much sense to allow a Model level specification of a DataLoader (at least for how the code operates now).  You are welcome to remove it completely from the Model object as part of this PR if you'd like.